server/py/services/api/api/endpoints/runs.py

```diff
@@ -518,3 +520,62 @@ async def abort_run(
     )
 
     return background_task
+
+
+@router.post(
+    "/projects/{project}/runs/{uid}/push_notifications",
+    response_model=mlrun.common.schemas.BackgroundTask,
+)
+async def push_notifications(
+    project: str,
+    uid: str,
+    background_tasks: BackgroundTasks,
+    auth_info: mlrun.common.schemas.AuthInfo = Depends(deps.authenticate_request),
+    db_session: Session = Depends(deps.get_db_session),
+    iter: int = 0,
+):
+    await (
+        framework.utils.auth.verifier.AuthVerifier().query_project_resource_permissions(
+            mlrun.common.schemas.AuthorizationResourceTypes.run,
+            project,
+            uid,
+            mlrun.common.schemas.AuthorizationAction.update,
+            auth_info,
+        )
+    )
+    run = await run_in_threadpool(
+        services.api.crud.Runs().get_run,
+        db_session,
+        uid,
+        iter,
+        project,
+        mlrun.common.formatters.RunFormat.notifications,
+    )
+
+    return framework.utils.background_tasks.ProjectBackgroundTasksHandler().create_background_task(
+        db_session,
+        project,
+        background_tasks,
+        _push_notification,
+        mlrun.mlconf.background_tasks.default_timeouts.push_notifications,
+        framework.utils.background_tasks.BackgroundTaskKinds.push_notification.format(
+            project, uid
+        ),
+        db_session,
+        run,
+    )
+
+
+async def _push_notification(db_session, run):
+    db = db_singleton.get_db()
+    framework.utils.notifications.unmask_notification_params_secret_on_task(
+        db, db_session, run
+    )
+    run_notification_pusher_class = (
+        framework.utils.notifications.notification_pusher.RunNotificationPusher
+    )
+    run_notification_pusher_object = run_notification_pusher_class(
+        [run],
+        run_notification_pusher_class.resolve_notifications_default_params(),
+    )
+    await run_in_threadpool(run_notification_pusher_object.push)
```

Comment:
this is all needs to run on a thread
look at the functions implementation, they are not async-awaitable